pkg/arm/resources.go

```diff
@@ -128,11 +128,14 @@ func fixupDepends(azProfile *api.AzProfile, template map[string]interface{}) {
 	}
 }
 
+// resourceID constructs an Azure resource ID offline from the provided components
 func resourceID(subscriptionID, resourceGroup, resourceProvider, resourceName string) string {
 	return fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/%s/%s", subscriptionID, resourceGroup, resourceProvider, resourceName)
 }
 
 func vnet(cs *api.OpenShiftManagedCluster) *network.VirtualNetwork {
+	// populating VnetID for customer's convenience
+	cs.Properties.NetworkProfile.VnetID = resourceID(cs.Properties.AzProfile.SubscriptionID, cs.Properties.AzProfile.ResourceGroup, "Microsoft.Network/virtualNetworks", vnetName)
```

Comment:
It doesn't look like this is the right place to be populating `cs`. Probably something that is better suited during conversion from external?

Reply:
Ok, modifying cs in the arm package is against intuition.
What would be the better solution:
exporting resourceID in the arm package?
moving ResourceID to a separate util since it's not ARM specific?

Reply:
Discussed on call with @kargakis. Will copy the function to the convert package for now. Created #978 to track that.

Reply:
let's catch up on this.  I think it can/should all be done in this PR (#978 not necessary).  Moving resourceID() to somewhere under util is fine.  I think the point where we modify cs actually needs to be a new step right at the bottom of `func (p *plugin) CreateOrUpdate`.

Reply:
moved to `func (p *plugin) CreateOrUpdate` added `pkg/util/resourceid` with only the `ResourceID` func.